packages/flutter/lib/src/material/switch.dart

```diff
@@ -1904,6 +1915,9 @@ class _SwitchConfigCupertino with _SwitchConfig {
   // Hand coded default based on the animation specs.
   @override
   double? get thumbOffset => null;
+
+  @override
+  Size get switchMinSize => const Size.square(kMinInteractiveDimension - 8.0);
```

Comment:
I wonder if ideally the switch should have a hit area of size kMinInteractiveDimension, even though visually it is not that big? I think the text selection handles do this for example, though it's easier for them because their size doesn't affect layout of other widgets.

Reply:
Though looking at your screenshots, it looks like the Switch doesn't get below 48x48 anyway?

Reply:
Good question! This confused me at first too but looks like min size and and switch width are used differently. 

> Though looking at your screenshots, it looks like the Switch doesn't get below 48x48 anyway?

Switch uses `effectiveMaterialTapTargetSize` with the `switchWidth` value for the hit area, not the `switchMinSize`.

https://github.com/flutter/flutter/blob/639910720cbce625c4f206ebe12e464c9217f627/packages/flutter/lib/src/material/switch.dart#L565-L571


While the the `switchMinSize` is used for the switch height and track length (which is smaller than the hit area).

https://github.com/flutter/flutter/blob/639910720cbce625c4f206ebe12e464c9217f627/packages/flutter/lib/src/material/switch.dart#L784-L792